pkg/clusterapi/proxy.go

```diff
@@ -8,12 +8,64 @@ import (
 	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
 
 	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/logger"
 	"github.com/aws/eks-anywhere/pkg/templater"
 )
 
 //go:embed config/http-proxy.conf
 var proxyConfig string
 
+func proxy(clusterSpec *cluster.Spec) bootstrapv1.ProxyConfiguration {
+	return bootstrapv1.ProxyConfiguration{
+		HTTPSProxy: clusterSpec.Cluster.Spec.ProxyConfiguration.HttpsProxy,
+		NoProxy:    noProxyList(clusterSpec.Cluster.Spec),
+	}
+}
+
+// SetProxyConfigInKubeadmControlPlane sets up proxy configuration in kubeadmControlPlane.
+// Currently only support proxy setting for bottlerocket and ubuntu. No-op for any other OS.
+func SetProxyConfigInKubeadmControlPlane(kcp *controlplanev1.KubeadmControlPlane, clusterSpec *cluster.Spec, os v1alpha1.OSFamily) error {
+	if clusterSpec.Cluster.Spec.ProxyConfiguration == nil {
+		return nil
+	}
+
+	switch os {
+	case v1alpha1.Bottlerocket:
+		kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.Proxy = proxy(clusterSpec)
+		kcp.Spec.KubeadmConfigSpec.JoinConfiguration.Proxy = proxy(clusterSpec)
+
+	case v1alpha1.Ubuntu:
+		return addProxyConfigInKubeadmConfigSpecFiles(&kcp.Spec.KubeadmConfigSpec, clusterSpec.Cluster.Spec)
+
+	default:
+		logger.MarkWarning(fmt.Sprintf("  Warning: unsupported OS family %s for proxy config. Please use %s or %s", os, v1alpha1.Bottlerocket, v1alpha1.Ubuntu))
```

Comment:
can we pass the logger as an input?
It makes code more reusable from the controller.
Also those `MarkWarning` methods only make sense in the CLI. And it proxy is not supported with a particular osFamily I would expect a validation error before we get here.